packages/devtools/lib/src/memory/memory.dart

```diff
@@ -185,6 +228,205 @@ class MemoryScreen extends Screen with SetStateMixin {
+        }
+        return;
+      }
+      row++;
+    }
+
+    framework.toast('Unable to find class $className', title: 'Error');
+  }
+
+  // Show the spinner we're computing...
+  Spinner createSpinner(Table<Object> table) {
+//    final html.Rectangle r = table.element.element.getBoundingClientRect();
+//    final int topPosition = r.top + (r.height / 2).round();
+//    final int leftPosition = r.left + (r.width / 2).round();
+
+    final Spinner spinner = Spinner.centered();
+//TODO(terry): Cleanup
+    //final Spinner spinner = Spinner()..clazz('padded');
+/*
+    spinner.element.style
+      ..position = 'absolute'
+      ..top = '${topPosition}px'
+      ..left = '${leftPosition}px'
+      ..zIndex = '1000';
+*/
+    return spinner;
+  }
+
+  Future<int> _selectInstanceInFieldHashCode(
+      String field, String hashCode) async {
+    final Table<Object> instanceTable = tableStack.elementAt(1);
+    final Spinner spinner = createSpinner(instanceTable);
+    instanceTable.element.add(spinner);
+
+    // There's an instances table up.
+    final List<InstanceSummary> instances = instanceTable.data;
+    int row = 0;
+    for (InstanceSummary instance in instances) {
+      // Check the field in each instance looking to find the object being held
+      // (the hashCode passed in matches the particular field's hashCode)
+      final evalResult = await evaluate(instance.objectRef, '$field.hashCode');
+      if (evalResult != null && evalResult.valueAsString == hashCode) {
+        // Found the object select the instance.
+        instanceTable.selectByIndex(row, scrollBehavior: 'auto');
+        spinner.element.remove();
+        return row;
+      }
+      row++;
+    }
+
+    spinner.element.remove();
+
+    framework.toast('Unable to find instance for field $field [$hashCode]',
+        title: 'Error');
+
+    return -1;
+  }
+
+  void _resetHistory() {
+    history..clazz('history-navigation-hidden', removeOthers: true);
+    history.clear();
+    memoryPath = NavigationPath();
+  }
+
+  /// Finish callback from search class selected (auto-complete).
+  void _callbackPopupSelectClass([bool cancel]) {
+    if (cancel) {
+      heapPopup.matcher.reset();
+      heapPopupList.reset();
+    } else {
+      // Reset memory history selecting a class.
+      _resetHistory();
+
+      // Highlighted class is the class to select.
+      final String classSelected = heapPopupList.highlightedItem;
+      _selectClass(classSelected);
+    }
+
+    // Done with the popup.
+    heapPopup.hide();
+  }
+
+  Future<void> _selectInstanceByHashCode(String objectHashCode) async {
+    // There's an instances table up.
+    final Table<Object> instanceTable = tableStack.last;
+    final List<InstanceSummary> instances = instanceTable.data;
+    int row = 0;
+    for (InstanceSummary instance in instances) {
+      // Check each instance looking to find a particular object.
+      final evalResult = await evaluate(instance.objectRef, 'hashCode');
```

Comment:
have you tested performance using a physical device instead of an emulator? I'm surprised this is as responsive as it is from the demos I've seen.

Reply:
I've been using an emulator and chromote to get worst UI lag and perf and its okay in worse scenario I'll be trying in best to see how nice it is :-) I hope?